src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicBuffers.cs

```diff
@@ -39,35 +39,94 @@ private void FreeNativeMemory()
         NativeMemory.Free(buffers);
     }
 
+    private void Reserve(int count)
+    {
+        if (_handles.Length < count)
```

Comment:
The size of `_buffers` is always the same as `_handles`, so they will either both grow or neither of them will.

`_count` contains the number of valid slots in those arrays, depending on history, it may be lower than `_handles.Length`